server/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java

```diff
@@ -968,6 +969,22 @@ protected Query doToQuery(SearchExecutionContext context) throws IOException {
         return query;
     }
 
+    @Override
+    protected QueryBuilder doCoordinatorRewrite(CoordinatorRewriteContext coordinatorRewriteContext) {
```

Comment:
I see that query_string does not override any of the rewrite methods yet. I was wondering how querying multiple fields is combined in a single lucene query depending on the operators, and I think that we can't state for sure that if one clause is on the tier field we can rewrite to match_none or match_all.

In general, the approach I would take is the following: if queries have doIndexMetadataRewrite with logic on constant field type, we can do the same in coordinating rewrite for the data tier field. Queries that don't do that based on mappings, shouldn't do it on the coord node either.

Reply:
Ah great shout! Sounds good, I'll reduce the scope here. Thanks !